accounts/store_test.go

```diff
@@ -430,6 +430,15 @@ func TestAccountUpdateMethods(t *testing.T) {
 		// error.
 		err = store.DeleteAccountPayment(ctx, acct.ID, hash1)
 		require.ErrorContains(t, err, "is not associated")
+
+		// Try once more to insert a payment that is currently unknown
+		// but this time add the WithErrIfUnknown option. This should
+		// return the ErrPaymentUnknown error.
+		_, err = store.UpsertAccountPayment(
+			ctx, acct.ID, hash1, 600, lnrpc.Payment_SUCCEEDED,
+			WithErrIfUnknown(),
+		)
+		require.ErrorIs(t, err, ErrPaymentUnknown)
```

Comment:
would it make sense to add a test that changes the status as well, since `removePayment` does that (perhaps using `WithErrIfUnknown` and `WithPendingAmount` together?

Reply:
the tests already covers changing the status. See above: the initial status  for `hash2` is Unknown, then it gets changed to `succeeded`

Reply:
gotcha 👍  - updated